std/experimental/allocator/package.d

```diff
@@ -1600,6 +1609,67 @@ unittest //bugzilla 15721
     Mallocator.instance.dispose(foo);
 }
 
+unittest //bugzilla 16824
+{
+    struct TestAllocator
+    {
+        import std.experimental.allocator.common: platformAlignment;
+        import std.experimental.allocator.mallocator: Mallocator;
+
+        alias allocator = Mallocator.instance;
+
+        private static struct ByteRange
+        {
+            void* ptr;
+            size_t length;
+        }
+
+        private ByteRange[] _allocations;
+
+        enum uint alignment = platformAlignment;
+
+        void[] allocate(size_t numBytes)
+        {
+             auto ret = allocator.allocate(numBytes);
+             _allocations ~= ByteRange(ret.ptr, ret.length);
+             return ret;
+        }
+
+        bool deallocate(void[] bytes)
+        {
+            import std.algorithm: remove, canFind;
+            import std.conv: text;
+
+            bool pred(ByteRange other)
+            { return other.ptr == bytes.ptr && other.length == bytes.length; }
+
+            assert(_allocations.canFind!pred);
+
+             _allocations = _allocations.remove!pred;
+             return allocator.deallocate(bytes);
+        }
+
+        ~this()
+        {
+            assert(!_allocations.length);
+        }
+    }
+
+    import std.experimental.allocator: dispose, makeArray;
+
+    TestAllocator allocator;
+
+    auto ints2d = allocator.makeArray!(int[][])(2);
+    foreach (ref ints1d; ints2d)
+        ints1d = allocator.makeArray!(int[])(3);
+
+    foreach (ref ints1d; ints2d)
+        foreach(ref ints0d; ints1d)
+            ints0d = allocator.makeArray!(int)(4);
```

Comment:
parenthesis aren't needed here and how about adding this to `foreach` loop before?